frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,327 @@
+      <MenuInput translate="true">
+        <TextInput
+          autoFocus
+          value={filterText}
+          aria-label={isProjects ? t('public~Select project...') : t('public~Select namespace...')}
+          iconVariant="search"
+          type="search"
+          placeholder={isProjects ? t('public~Select project...') : t('public~Select namespace...')}
+          onChange={(value: string) => setFilterText(value)}
+          className="co-namespace-selector__filter"
+          ref={filterRef}
+        />
+      </MenuInput>
+    </MenuGroup>
+  );
+
+  const isFavorite = (option) => (bookmarks?.[option.key] ? true : false); // undefined cannot be an option
+
+  const isOptionShown = (
+    option: {
+      title: string;
+      key: string;
+    },
+    checkIsFavorite?: boolean,
+  ) => {
+    const containsFilterText = option.title.toLowerCase().includes(filterText.toLowerCase());
+
+    return (
+      containsFilterText &&
+      (systemNamespacesInSelect || !isSystemNamespace(option)) &&
+      (!checkIsFavorite || isFavorite(option))
+    );
+  };
+  const noResults = (
+    <>
+      <Divider />
+      <EmptyState>
+        <Title size="md" headingLevel="h4">
+          {isProjects ? t('public~No projects found') : t('public~No namespaces found')}
+        </Title>
+        <EmptyStateBody>{t('public~No results match the filter criteria.')}</EmptyStateBody>
+        <EmptyStatePrimary>
+          <Button
+            variant="link"
+            onClick={(event) => {
+              event.preventDefault();
+              event.stopPropagation();
+              setFilterText('');
+              filterRef.current?.focus();
+            }}
+            className="co-namespace-selector__clear-filters"
+          >
+            {t('Clear filters')}
+          </Button>
+        </EmptyStatePrimary>
+      </EmptyState>
+    </>
+  );
+
+  const filteredOptions = (favorites?: boolean) =>
+    options.filter((option) => isOptionShown(option, favorites));
+
+  const menuGroup = (favorites?: boolean) => {
+    const label = favorites ? t('public~Favorites') : t('public~Projects');
+
+    if (filteredOptions(favorites).length === 0) {
+      return favorites ? null : noResults;
+    }
+
+    const menuItems = filteredOptions(favorites).map((option) => {
+      return (
+        <MenuItem
+          translate="true"
+          key={option.key}
+          itemId={option.key}
+          isFavorited={isFavorite(option)}
+          isSelected={selected === option.key}
+          isActive={activeOption === option.key}
+        >
+          {option.title}
+        </MenuItem>
+      );
+    });
+    return (
+      <>
+        <Divider />
+        <MenuGroup label={label} translate="true">
+          <MenuContent translate="true">
+            <MenuList translate="true">{menuItems}</MenuList>
+          </MenuContent>
+        </MenuGroup>
+      </>
+    );
+  };
+
+  const actionList = (
+    <MenuList translate="true">
```

Comment:
When we remove all of the logic for the legacy create action handling, we should be able to just use the MenuFooter component and remove this MenuList wrapper.
```suggestion
```